documentation/development-docs/design-guidelines/cmdlet-best-practices.md

```diff
@@ -0,0 +1,160 @@
+# Cmdlet Best Practices
+
+### Cmdlet Naming Conventions
+
+The following are naming conventions to keep in mind when coming up with a name for your cmdlet.
+
+#### Verb-Noun Format
+
+Cmdlet names should follow the _Verb-Noun_ format, where the verb is from the [list of approved PowerShell verbs](https://docs.microsoft.com/en-us/powershell/developer/cmdlet/approved-verbs-for-windows-powershell-commands), and the noun is a specific noun describing a resource within your service.
+
+#### Noun Prefix
+
+For ARM cmdlets, the noun must be prefixed with `Az`.
+
+#### Pascal Case
+
+From the [_Strongly Encouraged Development Guidelines_](https://docs.microsoft.com/en-us/powershell/developer/cmdlet/strongly-encouraged-development-guidelines#use-pascal-case-for-cmdlet-names-sd02):
+
+> _Use Pascal case for cmdlet names. In other words, capitalize the first letter of the verb and all terms used in the noun. For example, "Clear-ItemProperty"._
+
+#### Specific Noun and Noun Singularity
+
+From the [_Strongly Encouraged Development Guidelines_](https://docs.microsoft.com/en-us/powershell/developer/cmdlet/strongly-encouraged-development-guidelines#use-a-specific-noun-for-a-cmdlet-name-sd01):
+
+> _Nouns used in cmdlet naming need to be very specific so that the user can discover your cmdlets. Prefix generic nouns such as "server" with a shortened version of the product name. For example, if a noun refers to a server that is running an instance of Microsoft SQL Server, use a noun such as "SQLServer". The combination of specific nouns and the short list of approved verbs enable the user to quickly discover and anticipate functionality while avoiding duplication among cmdlet names._
+>
+> _To enhance the user experience, the noun that you choose for a cmdlet name should be singular. For example, use the name `Get-Process` instead of `Get-Processes`. It is best to follow this rule for all cmdlet names, even when a cmdlet is likely to act upon more than one item._
+
```

Comment:
Should we consider a noun that sounds weird, like  `Settings`,  as an exception?
`Update-Setting` vs `Update-Settings`

Reply:
I don't think we should call it out in this doc -- we should advise using singular nouns in all cases, and we'll handle special cases (_e.g._, weird sounding nouns) during the design review.